src/helper/remote_config/protocol/config_state.hpp

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#pragma once
+
+#include <string>
+
+namespace dds::remote_config::protocol {
+
+struct config_state {
+public:
+    config_state(std::string &id, int version, std::string &product)
+        : _id(id), _version(version), _product(product){};
+    std::string get_id() { return _id; };
```

Comment:
```suggestion
    std::string get_id() const { return _id; };
```
I don't think this should return a copy but rather a const reference. If the caller needs a copy it's trivial enough to do so.

Reply:
Done in all getters